### PR TITLE
B 17788 get table rest

### PR DESCRIPTION
**Pull request by cameroncaci**
[Ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a857401)

# Discussion
Please see the testing section as I have pointed out an area for conversation within this PR.

# Changes
* Refactored GetTable section of the code to form new REST calls based on the `GatewayService`.
* Changed TRDM package file structure and organization to try and improve readability. Such as moving the flow start to a `trdm.go` file and instead of the function being called `LastTableUpdate` it's now called `BeginTGETFlow` that creates the cron job which calls lastTableUpdate.
* Made some minor touch ups to lastTableUpdate such as logger locations, variables, and the calling and handling of getTable.

# Review
Please see documentation [here](https://aws.github.io/aws-sdk-go-v2/docs/) and [here](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2) to provide better feedback.

# Testing
The code is now at a point of testing - though interestingly I've encountered a little hiccup when attempting to do so manually. In theory, as long as the server boots up with staging credentials it should attempt to call `BeginTGETFlow`. As of right now, I'm encountering a little issue with the AWS credentials provided, this may be due to the server automatically using dev creds and not being able to get overwritten with stg creds.  It is failing on my end due to having an empty `'roleArn'`. I would encourage this as a topic of discussion within the PR.

If we can't test it properly within local, we may push it up to staging for it to just fail there. What should happen is on the local side you should be able to provide staging credentials (Such as using `aws-vault`), turn on the server, and it should be able to create a proper provider, form a good request, sign that request, and then attempt to send it to the gateway. However, right now it's failing due to creds before even sending it off.

**NOTE**
It **_should_** fail. But, we should get denied at gateway entry not on credential signage. See security reasons why it would fail locally [here](https://dp3.atlassian.net/wiki/spaces/MT/pages/2275573761/TRDM+Soap+Proxy+API+Gateway+Lambda+Function).

- To wrap your server in the staging credentials run this command: `aws-vault exec ${AWS_PROFILE} -- make server_run`

# Steps for Reviewers

- [ ] Ensure the server runs normally during local development
- [ ] Ensure the server can turn on normally when wrapped in staging credentials, and that it can sign its request and send off to the API gateway. It should receive a failing response in return, but it should still be able to send off properly.

### CODE REVIEW

**Comment by cameroncaci** [comment]
The empty role arn could be due to something local.. still trying to discover more

**Comment by cameroncaci** [comment]
There's a chance it may already be g2g, but want to verify all is good first of course

File: pkg/trdm/api_gateway.go [hunk 1]
```diff
@@ -79,7 +81,46 @@ func (gs GatewayService) gatewayLastTableUpdate(request models.LastTableUpdateRe
...
+	// Generate a SHA256 hash for signing
+	hash := generateSHA256Hash(requestBody)
+
+	// Put it into a new request
+	req, err := http.NewRequest("POST", gs.gatewayURL+GetTableEndpoint, bytes.NewBuffer(requestBody))
+	if err != nil {
+		gs.logger.Error("getTable request", zap.Error(err))
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
```
[the remark this answers is not in the document]
Comment (cameroncaci) [reply pkg/trdm/api_gateway.go:101]: Unfortunately Go's HTTP package does not have a pre-built constant to apply for application/json. As using the string literal is common throughout Go and expected usage per their docs I would recommend to leave as is in our codebase

**Review by danieljordan-caci — APPROVED** [review]
LGTM! Thanks for putting in comments.

File: cmd/milmove/serve.go [hunk 2]
```diff
@@ -836,7 +839,7 @@ func serveFunction(cmd *cobra.Command, args []string) error {
 		httpClient := &http.Client{Transport: tr, Timeout: time.Duration(30) * time.Second}
 
 		// Initial REST call for LastTableUpdate on server start and once per day
```
Comment (traskowskycaci) [inline cmd/milmove/serve.go:841]: Do you want to update LastTableUpdate in the comment here, and is the "on server start and once per day" still accurate?

**Review by traskowskycaci — APPROVED** [review]

File: cmd/milmove/serve.go (hunk 2 above)
Comment (cameroncaci) [reply cmd/milmove/serve.go:841]: Good catch, thank you. It does indeed last a `lastTableUpdate` REST call; however, I'll make the comment more descriptive. And yes it does run on server start and also once per day (At midnight)

**Pull request merged**